### DISCUSSION TITLE
How to use all free height for div?

### DISCUSSION THREAD

**Discussion by LWJerri** [tailwindlabs/tailwindcss#8678] [Help] [answered]
Hello everyone! I wanna use all free height space for div, but `h-full` or `h-screen` didn't help me, because element in `div.div.div...`. 

Very important for me to add a scrollbar if all elements can be displayed in div.

Currently i have:
![image](https://user-images.githubusercontent.com/50290430/174489665-ae854bf6-36a3-4a8f-a3bc-33ae9b4ecb49.png)
![image](https://user-images.githubusercontent.com/50290430/174489675-1b771168-3ecb-41c7-b657-ecd44fdf0370.png)

But I wanna make it:
![image](https://user-images.githubusercontent.com/50290430/174489714-78f5c278-bc9a-406e-bf66-a54a328a30f4.png)

Minimal code to reproduce my problem:
```vue
<div class="w-40 rounded border-r border-b border-gray-700">
      <button>+</button>
      <div class="form-floating">
        <input placeholder="command">
        <label>test</label>
      </div> <!--75vh-->
      <ul
        class="menu scrollbar-thin overflow-auto scrollbar scrollbar-thumb-gray-900 scrollbar-track-gray-600"
      >
        <li>TEXT</li>
      </ul>
    </div>
```

**Comment by crswll** [comment, marked as answer]
You'd typically want to use `flex flex-col` and use `flex-1` on a direct child element that you want to fill up the remaining space.

Here's a really minimal example: https://play.tailwindcss.com/jMyVK5Jc5P
